crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -632,24 +632,36 @@ impl CumulocityConverter {
         &mut self,
         message: &MqttMessage,
     ) -> Result<Vec<MqttMessage>, ConversionError> {
-        let operation = C8yOperation::from_json(message.payload.as_str()?)?;
-        let device_xid = operation.external_source.external_id;
-        let cmd_id = self.command_id.new_id_with_str(&operation.op_id);
+        // Smartrest messages on c8y/devicecontrol/notifications can contain multiple operations in a single MQTT
+        // message, so split them
+        let operation_payloads = collect_smartrest_messages(message.payload_str()?);
```

Comment:
I was surprised that `collect_smartrest_messages()` works here. However, after seeing the contents of the function, it makes sense. That means, the function should be renamed now, otherwise it's very misleading. Maybe `collect_c8y_messages()` or so on.

Reply:
Right, it's not Smartrest, so probably it would also be fine to just split on `\n`, right? As JSON over MQTT doesn't have these escaping problems the function was originally made for?